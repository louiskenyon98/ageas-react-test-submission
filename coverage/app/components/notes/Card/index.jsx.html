<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/notes/Card/index.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\index.html">All files</a> / <a href="index.html">app/components/notes/Card</a> index.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/2</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import {Link} from 'react-router-dom';
&nbsp;
import styles from '../../../../themes/style.scss';
&nbsp;
class Card extends React.PureComponent {
    render() {
        return (
            &lt;div className={`col-xs-1 col-md-4 col-xl-3 d-flex align-items-stretch ${styles.column}`}&gt;
                &lt;div className={`ui raised card ${styles.card}`}&gt;
                    &lt;div className={`content`}&gt;
                        &lt;i
                            // onClick={() =&gt; this.props.delete(this.props.id)}
                            onClick={() =&gt; this.props.delete(this.props.id)}
                            className={`right floated close icon ${styles["delete-button"]}`}
                        /&gt;
                        &lt;div className={`header ${styles.header}`}&gt;
                            {this.props.title}
                        &lt;/div&gt;
                        &lt;div className={`description ${styles.body}`}&gt;
                            &lt;p&gt;
                                {this.props.body}
                            &lt;/p&gt;
                        &lt;/div&gt;
                    &lt;/div&gt;
                    &lt;div className={"extra content"}&gt;
                        &lt;span className="right floated pencil alternate"&gt;
                            {/*pass props to link*/}
                            &lt;Link to={`/edit/${this.props.id}`}&gt;
                                &lt;i className={`pencil alternate icon ${styles["edit-button"]}`}/&gt;
                            &lt;/Link&gt;
                        &lt;/span&gt;
                    &lt;/div&gt;
                &lt;/div&gt;
            &lt;/div&gt;
        )
    }
}
&nbsp;
export default Card;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Apr 16 2019 09:43:43 GMT+0100 (GMT Daylight Time)
</div>
</div>
<script src="..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\sorter.js"></script>
</body>
</html>
