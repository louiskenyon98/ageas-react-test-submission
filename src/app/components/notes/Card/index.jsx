import React from 'react';
import {Link} from 'react-router-dom';

import styles from '../../../../themes/style.scss';

class Card extends React.PureComponent {
    render() {
        return (
            <div className={`col-xs-1 col-md-4 col-xl-3 d-flex align-items-stretch ${styles.column}`} key={this.props.id}>
                <div className={`ui raised card ${styles.card}`}>
                    <div className={`content`}>
                        <i
                            onClick={() => this.props.deleteFunc(this.props.id)}
                            className={`right floated close icon ${styles["delete-button"]}`}
                        />
                        <div className={`header ${styles.header}`}>
                            {this.props.title}
                        </div>
                        <div className={`description ${styles.body}`}>
                            <p>
                                {this.props.body}
                            </p>
                        </div>
                    </div>
                    <div className={"extra content"}>
                        <span className="right floated pencil alternate">
                            {/*pass props to link*/}
                            <Link to={`/edit/${this.props.id}`}>
                                <i className={`pencil alternate icon ${styles["edit-button"]}`}/>
                            </Link>
                        </span>
                    </div>
                </div>
            </div>
        )
    }
}

export default Card;
