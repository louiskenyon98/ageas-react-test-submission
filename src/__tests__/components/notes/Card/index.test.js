import React from 'react';
import {shallow} from 'enzyme';
import Card from '../../../../app/components/notes/Card/index';

describe('Card', () => {
    let props = {};
    let wrapper = () => shallow(<Card {...props}/>);
    beforeEach(() => {
        props ={
            id: 1,
            title: 'noteCard Title',
            body: 'noteCard Body',
            delete: jest.fn()
        }
    });

    describe('render', () => {
        it('should render a card with a title, body, and icons', () => {
            expect(wrapper()).toMatchSnapshot();
        });
    });
    describe('functionality', () => {
        it('should call delete onClick', () => {
            expect(props.delete).not.toHaveBeenCalled();
            wrapper().find('i').first().simulate('click');
            expect(props.delete).toHaveBeenCalledTimes(1);
        })
    })

});
