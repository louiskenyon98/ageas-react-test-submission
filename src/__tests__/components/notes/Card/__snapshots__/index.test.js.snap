// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card render should render a card with a title, body, and icons 1`] = `
<div
  className="col-xs-1 col-md-4 col-xl-3 d-flex align-items-stretch column"
>
  <div
    className="ui raised card card"
  >
    <div
      className="content"
    >
      <i
        className="right floated close icon delete-button"
        onClick={[Function]}
      />
      <div
        className="header header"
      >
        noteCard Title
      </div>
      <div
        className="description body"
      >
        <p>
          noteCard Body
        </p>
      </div>
    </div>
    <div
      className="extra content"
    >
      <span
        className="right floated pencil alternate"
      >
        <Link
          replace={false}
          to="/edit/1"
        >
          <i
            className="pencil alternate icon edit-button"
          />
        </Link>
      </span>
    </div>
  </div>
</div>
`;
